llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -5250,10 +5250,15 @@ unsigned SIInstrInfo::getVALUOp(const MachineInstr &MI) const {
   case AMDGPU::S_FLOOR_F32: return AMDGPU::V_FLOOR_F32_e64;
   case AMDGPU::S_TRUNC_F32: return AMDGPU::V_TRUNC_F32_e64;
   case AMDGPU::S_RNDNE_F32: return AMDGPU::V_RNDNE_F32_e64;
-  case AMDGPU::S_CEIL_F16: return AMDGPU::V_CEIL_F16_t16_e64;
-  case AMDGPU::S_FLOOR_F16: return AMDGPU::V_FLOOR_F16_t16_e64;
-  case AMDGPU::S_TRUNC_F16: return AMDGPU::V_TRUNC_F16_t16_e64;
-  case AMDGPU::S_RNDNE_F16: return AMDGPU::V_RNDNE_F16_t16_e64;
+  case AMDGPU::S_CEIL_F16:
+    return ST.useRealTrue16Insts() ? AMDGPU::V_CEIL_F16_t16_e64
+                                   : AMDGPU::V_CEIL_F16_fake16_e64;
```

Comment:
@mbrkusanin Mirko, we don't seem to have any coverage for this `S_CEIL_F16` case. Should we add some?

Reply:
Not every instruction is covered, only one per group. "cmp_f16" test from llvm/test/CodeGen/AMDGPU/fix-sgpr-copies-f16.mir was supposed to cover s_*_f16 insts.
For the fake16 everything should work, but for t16 regclass is different, so this change is not enough. So, yeah we should have at least one test for the t16 ones.

Reply:
Thanks. I've added a test and updated the S_* -> V_* translation to handle the case.